harness/determined/common/api/authentication.py

```diff
@@ -22,6 +23,23 @@ def salt_and_hash(password: str) -> str:
         return password
 
 
+def check_password_complexity(password: str) -> None:
```

Comment:
If this is meant to be a public function, then I think the docstring needs to describe the checks. An external user shouldn't have to read code to understand what a function is doing.

If it's meant to be package-private (see [the style guide](https://hpe-aiatscale.atlassian.net/wiki/spaces/ENGINEERIN/pages/1505230999/Python+Code+Style#Rule-1%3A-Public-and-private-modules-and-symbols) for more details), then this documentation is good enough. If it's meant to be package-private, though, then this should not be re-exported in `common/api/__init__.py`.

Reply:
Thanks! That's a good docstring.

When you see this and are up for a chat, could we please talk about `check_password_complexity` and the plusses/minuses about making it public or package-private? I'm interested in how you want users to use it both in today's universe and in one where DET-10209 is implemented.